src/main.cpp

```diff
@@ -5689,23 +5728,36 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
         CBloomFilter filterMemPool;
         CInv inv;
         vRecv >> inv >> filterMemPool;
+        if (!((inv.type == MSG_XTHINBLOCK)||(inv.type == MSG_THINBLOCK)))
```

Comment:
This was my original intention which is why you see my doc comment in "CInv::IsKnownType" (I had a few addtl helper functions defined).  However since each helper function was used just once, I became convinced that doing so made the P2P receiver code less clear.  The purpose of these statements is to constrain an validate the passed CInv to the values allowed by this P2P message and to do so as soon as the inv is read to make the code easy to review.  Burying the actual CInv types allowed by this P2P message inside a helper function does not aid in clarity.